common/lib/capa/capa/inputtypes.py

```diff
@@ -232,7 +234,7 @@ def __init__(self, system, xml, state):
 
         # put hint above msg if it should be displayed
         if self.hintmode == 'always':
-            self.msg = self.hint + ('<br/>' if self.msg else '') + self.msg
+            self.msg = HTML('{hint}<br/>{msg}' if self.msg else '').format(hint=self.hint, msg=self.msg)
```

Comment:
Whether you use HTML() or Text(), if you call `.format()` afterwards you are going to HTML-escape what is being formatted in unless it is marked with `HTML()`.  So, if both of these could contain HTML, you would need:
```
self.msg = HTML('{hint}<br/>{msg}' if self.msg else '').format(
    hint=HTML(self.hint), 
    msg=HTML(self.msg)
)
```